### ISSUE TITLE
Timeline reverse, then play briefly causes wrong style

### ISSUE THREAD

**Issue by Clindbergh** [juliangarnier/anime#503]
**Describe the bug**
When starting a timeline, then reversing it after it played and playing it again, a wrong style is applied for a brief period.

**To Reproduce**

```
let inTimeline = anime.timeline({
  autoplay: false
});

inTimeline
  .add({
    targets: ".a1",
    opacity: [0, 1],
    duration: 200,
    easing: "linear"
  })
  .add({
    targets: ".a2",
    opacity: [0, 1],
    delay: anime.stagger(100),
    ease: "inOutElastic"
  });

document.getElementById("trigger").addEventListener(
  "click",
  function(e) {
    inTimeline.play();
    inTimeline.finished.then(() => {
      inTimeline.reverse();
    });
  },
  false
);
```

1. Click `#trigger`
2. Wait until the animation is finished
3. Click `#trigger` again

Result: For a short time `.a1` and `.a2` will have opacity 0, before having opacity 1 again and the fade out effect starts.

**Expected behavior**
Opacity stays at 1.

**Additional context**
Perhaps related: #373

**Reason:**
* After `inTimeline` is finished the first time, `inTimeline.reverse()` is called. This causes `inTimeline` and all its children to have the `reversed = true` ([index.js:925](https://github.com/juliangarnier/anime/blob/master/src/index.js#L925)). Notice that for both `inTimeline` and its children `completed` is `true`.
* When `inTimeline.play()` is called the second time, `inTimeline` and all its children are reset ([index.js:1123](https://github.com/juliangarnier/anime/blob/875a3d6745d4bff2e4c6ffe0e2d24aac3bf8a45a/src/index.js#L1123)), because `completed` was `true`. This causes the initial styles (opacity = 0) to be applied ([index.js:1096](https://github.com/juliangarnier/anime/blob/875a3d6745d4bff2e4c6ffe0e2d24aac3bf8a45a/src/index.js#L1096)).

Most likely [`toggleInstanceDirection()`](https://github.com/juliangarnier/anime/blob/875a3d6745d4bff2e4c6ffe0e2d24aac3bf8a45a/src/index.js#L920) is incompletely handling the change. Intuitively `completed` and `began` should be set correctly, e.g. when `completed` was `true`, it should become `false` and `began` should also become `false`. However I am not currently aware of all side effects and other things to consider.

**Comment by williambout** [comment]
I can repro as well

**Comment by hunterpollpeter** [comment]
Experiencing this issue as well.

**Comment by kn0wn** [comment]
Should be fixed in 3.2.0
